sdk/daml-lf/engine/src/test/scala/com/digitalasset/daml/lf/engine/EngineValidatePackagesTest.scala

```diff
@@ -19,17 +22,61 @@ class EngineValidatePackagesTest(majorLanguageVersion: LanguageMajorVersion)
     with Matchers
     with Inside {
 
-  val pkgId = Ref.PackageId.assertFromString("-pkg-")
-
   val langVersion = LanguageVersion.defaultOrLatestStable(majorLanguageVersion)
 
+  val pkgId = Ref.PackageId.assertFromString("-pkg-")
+  val extraPkgId = Ref.PackageId.assertFromString("-extra-")
+  val missingPkgId = Ref.PackageId.assertFromString("-missing-")
+  val utilityPkgId = Ref.PackageId.assertFromString("-utility-")
+  val altUtilityPkgId = Ref.PackageId.assertFromString("-alt-utility-")
+  val (stablePkgId, stablePkg) = StablePackages(majorLanguageVersion).packagesMap.head
+
   implicit val parserParameters: parser.ParserParameters[this.type] =
     parser.ParserParameters(pkgId, langVersion)
 
+  val fakeDamlPrimPkg =
+    p"""
+      metadata ( 'daml-prim' : '1.0.0' )
+      module Mod {
+        val string: Text = "fake-daml-prim";
+      }
+    """
+  val fakeDamlStdlibPkg =
+    p"""
+      metadata ( 'daml-stdlib' : '1.0.0' )
+      module Mod {
+        val string: Text = "fake-daml-stdlib";
+      }
+    """
+  val utilityPkgChoices = Seq(
+    ("no utility packages", Set.empty, Seq.empty),
+    ("daml-prim utility package", Set(utilityPkgId), Seq(utilityPkgId -> fakeDamlPrimPkg)),
+    (
+      "daml-stdlib utility package",
+      Set(altUtilityPkgId),
+      Seq(altUtilityPkgId -> fakeDamlStdlibPkg),
+    ),
+    (
+      "daml-prim and daml-stdlib utility package",
+      Set(utilityPkgId, altUtilityPkgId),
+      Seq(utilityPkgId -> fakeDamlPrimPkg, altUtilityPkgId -> fakeDamlStdlibPkg),
+    ),
+  )
+  val stablePkgChoices = Seq(
+    ("no stable packages", Set.empty, Seq.empty),
+    ("stable package", Set(stablePkgId), Seq(stablePkgId -> stablePkg)),
+  )
+
   private def newEngine = new Engine(
     EngineConfig(LanguageVersion.AllVersions(majorLanguageVersion))
   )
 
+  private def darFromPackageMap(
+      mainPkg: (Ref.PackageId, Package),
+      dependentPkgs: (Ref.PackageId, Package)*
+  ): Dar[(Ref.PackageId, Package)] =
+    Dar(mainPkg, dependentPkgs.toList)
+
   "Engine.validatePackages" should {
```

Comment:
Intention is that we will eventually be able to drop the `Engine.validatePackages` tests (in favour of the `Engine.validateDar` ones)